sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantCast.cs

```diff
@@ -42,5 +42,30 @@ public void N(int[,] numbers)
         {
             numbers.Cast<int>().Where(x => x > 0); // Compliant, multidimensional arrays need to be cast
         }
+
+        // https://github.com/SonarSource/sonar-dotnet/issues/3273
+        public void OfTypeWithReferenceTypes()
+        {
+            var stringArrayWithNull = new string[] { "one", "two", null, "three" };
+            var filteredStringArray = stringArrayWithNull.OfType<string>(); // Compliant, has 3 items, without 'null'
+
+            var enumerableOfObjects = new object[] { 1, "one", null };
+            var filteredObjectArray = enumerableOfObjects.OfType<object>(); // Compliant, has 2 items, without 'null'
+
+            var enumerableOfNullableInt = new int?[] { 1, 2, null };
+            var filteredInts = enumerableOfNullableInt.OfType<int?>(); // Compliant, has 2 items, without 'null'
```

Comment:
For the sake of completeness we can add also a FN  with an array of strings containing only non null values.

Reply:
Yes, but it will document the fact that we don't check for nullable or value types.